octodns/provider/googlecloud.py

```diff
@@ -0,0 +1,363 @@
+    """Wrapper to make google cloud client resource record sets from OctoDNS
+    Records.
+
+        googlecloud.py:
+        class: octodns.provider.googlecloud._GoogleCloudRecordSetMaker
+        An _GoogleCloudRecordSetMaker creates google cloued client resource
+        records which can be used to update the Google Cloud DNS zones.
+    """
+
+    def __init__(self, gcloud_zone, record):
+        self.gcloud_zone = gcloud_zone
+        self.record = record
+
+        self._record_set_func = getattr(
+            self, '_record_set_from_{}'.format(record._type))
+
+    def get_record_set(self):
+        return self._record_set_func(self.record)
+
+    def _record_set_from_A(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, record.values)
+
+    _record_set_from_AAAA = _record_set_from_A
+
+    def _record_set_from_CAA(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{flags} {tag} {value}'.format(**record.data['value'])])
+
+    def _record_set_from_CNAME(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [record.value])
+
+    def _record_set_from_MX(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{preference} {exchange}'.format(**v.data)
+                for v in record.values])
+
+    def _record_set_from_NAPTR(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{order} {preference} "{flags}" "{service}" '
+                '"{regexp}" {replacement}'
+                .format(**v.data) for v in record.values])
+
+    _record_set_from_NS = _record_set_from_A
+
+    _record_set_from_PTR = _record_set_from_CNAME
+
+    _record_set_from_SPF = _record_set_from_A
+
+    def _record_set_from_SRV(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{priority} {weight} {port} {target}'
+                .format(**v.data) for v in record.values])
+
+    def _record_set_from_TXT(self, record):
+        if 'values' in record.data:
+            val = record.data['values']
+        else:
+            val = [record.data['value']]
+
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, val)
+
+
+class GoogleCloudProvider(BaseProvider):
+    """
+    Google Cloud DNS provider
+
+    google_cloud:
+        class: octodns.provider.googlecloud.GoogleCloudProvider
+        # Credentials file for a service_account or other account can be
+        # specified with the GOOGLE_APPLICATION_CREDENTIALS environment
+        # variable. (https://console.cloud.google.com/apis/credentials)
```

Comment:
If possible octoDNS prefers to provide credentials explicitly with config vars and params to clients rather than using them from the ENV. That generally allows for the management of records in multiple accounts in the same config/setup. E.g. if you worked at a company with multiple Google Cloud accounts you could configure something like:

```yaml
providers:
  google_cloud_us:
    ...
  google_cloud_eu:
    ...
```

Reply:
Yes. Updated this too.